resources/puppetlabs/lein-ezbake/template/global/ext/cli/start.erb

```diff
@@ -1,21 +1,59 @@
 #!/usr/bin/env bash
+pid=$(pgrep -f <%= EZBake::Config[:uberjar_name] %>)
+
 set -e
 
 restartfile=<%= EZBake::Config[:restart_file] %>
-timeout=<%= EZBake::Config[:start_timeout] %>
+start_timeout=<%= EZBake::Config[:start_timeout] %>
+stop_timeout=60
 dir=$(dirname "$restartfile")
 realname="<%= EZBake::Config[:real_name] %>"
 PIDFILE="/var/run/puppetlabs/${realname}/${realname}.pid"
 
-find_my_pid() {
-    pid=`pgrep -f <%= EZBake::Config[:uberjar_name] %>`
-    if [ ! -d  "/var/run/puppetlabs/${realname}" ] ; then
+write_pid_file() {
+    if [ ! -d "/var/run/puppetlabs/${realname}" ]; then
         mkdir -p /var/run/puppetlabs/${realname}
         chown -R $USER:$GROUP /var/run/puppetlabs/${realname}
     fi
-    echo $pid > $PIDFILE
+    echo $1 > $PIDFILE
 }
 
+terminate_java_process() {
+    set +e
+
+    echo "Startup script was terminated before completion" 1>&2
+    kill -TERM $pid >/dev/null 2>&1
+    sleep 0.1
+
+    timeout=$stop_timeout
+    kill -0 $pid >/dev/null 2>&1
+    while [ $? == 0 ] && [ "$timeout" != 0 ]; do
```

Comment:
you've got a mix of `== 0` and `= 0` here. Would recommend `-eq 0` and `ne 0` for integer comparisons.